### PR TITLE
docs: roll 1.2.0

### PR DESCRIPTION
**Pull request by smol-ninja**
### Changelog
- Closes https://github.com/sablier-labs/v2-periphery/issues/306
- Closes https://github.com/sablier-labs/v2-periphery/issues/352

### Subtasks
- [x] https://github.com/sablier-labs/v2-periphery/pull/355
- [x] https://github.com/sablier-labs/v2-periphery/discussions/311#discussioncomment-9086475

### CODE REVIEW

**Comment by PaulRBerg** [comment]
As discussed previously, we will review and merge this PR after the private audit is finished.

**Comment by smol-ninja** [comment]
This is also ready for review @andreivladbrg.

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,13 +4,38 @@ All notable changes to this project will be documented in this file.
...
+  [#300](https://github.com/sablier-labs/v2-periphery/pull/300))
+- Refactor `Range` to `Timestamps` ([#335](https://github.com/sablier-labs/v2-periphery/pull/335))
+- Bump `sablier/v2-core` to v1.2.0
+- Bump OpenZeppelin to v5 ([#283](https://github.com/sablier-labs/v2-periphery/pull/283))
+- Bump build dependencies ([#351](https://github.com/sablier-labs/v2-periphery/pull/351))
+- Switch to Bun ([#249](https://github.com/sablier-labs/v2-periphery/pull/249))
+
+### Added
+
+- Add `MerkleLockup` and `BatchLockup` support for `LockupTranched`
```
Comment (andreivladbrg) [inline CHANGELOG.md:31]: let's add two different points, one for `MerkleLockup`, and one for batch functions, also PR references are missing

**Review by andreivladbrg — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (smol-ninja) [reply CHANGELOG.md]: Good point. Addressed in https://github.com/sablier-labs/v2-periphery/pull/329/commits/b56d5fe0cb98bf7f0badb0adcce42796fa52ef48.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,13 +4,39 @@ All notable changes to this project will be documented in this file.
...
+  [#300](https://github.com/sablier-labs/v2-periphery/pull/300))
+- Refactor `Range` to `Timestamps` ([#335](https://github.com/sablier-labs/v2-periphery/pull/335))
+- Bump `sablier/v2-core` to v1.2.0
+- Bump build dependencies ([#351](https://github.com/sablier-labs/v2-periphery/pull/351),
+  [#283](https://github.com/sablier-labs/v2-periphery/pull/283))
+- Switch to Bun ([#249](https://github.com/sablier-labs/v2-periphery/pull/249))
+
+### Added
+
+- Add `MerkleLockup` support for `LockupTranched` ([#297](https://github.com/sablier-labs/v2-periphery/pull/297))
```
Comment (andreivladbrg) [inline CHANGELOG.md:31]: should we also include https://github.com/sablier-labs/v2-periphery/pull/357 here?

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,13 +4,39 @@ All notable changes to this project will be documented in this file.
...
+- Rename `MerkleStreamer` to `MerkleLockup` ([#268](https://github.com/sablier-labs/v2-periphery/pull/268))
+- Rename `Batch` to `BatchLockup` ([#322](https://github.com/sablier-labs/v2-periphery/pull/322),
+  [#297](https://github.com/sablier-labs/v2-periphery/pull/297),
+  [#300](https://github.com/sablier-labs/v2-periphery/pull/300))
+- Refactor `Range` to `Timestamps` ([#335](https://github.com/sablier-labs/v2-periphery/pull/335))
+- Bump `sablier/v2-core` to v1.2.0
+- Bump build dependencies ([#351](https://github.com/sablier-labs/v2-periphery/pull/351),
+  [#283](https://github.com/sablier-labs/v2-periphery/pull/283))
+- Switch to Bun ([#249](https://github.com/sablier-labs/v2-periphery/pull/249))
+
```
Comment (andreivladbrg) [inline CHANGELOG.md:28]: let's add:
 - Use Solidity v0.8.26 https://github.com/sablier-labs/v2-periphery/pull/351

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,13 +4,39 @@ All notable changes to this project will be documented in this file.
...
 [1.0.0]: https://github.com/sablier-labs/v2-periphery/releases/tag/v1.0.0
 
+## [1.2.0] - 2024-07-01
+
+### Changed
+
+- Rename `MerkleStreamer` to `MerkleLockup` ([#268](https://github.com/sablier-labs/v2-periphery/pull/268))
+- Rename `Batch` to `BatchLockup` ([#322](https://github.com/sablier-labs/v2-periphery/pull/322),
+  [#297](https://github.com/sablier-labs/v2-periphery/pull/297),
+  [#300](https://github.com/sablier-labs/v2-periphery/pull/300))
```
Comment (andreivladbrg) [inline CHANGELOG.md:22]: why are #297 and #300 added at this point?

wouldn't #322 be suffice?

**Review by andreivladbrg — CHANGES_REQUESTED** [review]
left some comments, see below:

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,13 +4,39 @@ All notable changes to this project will be documented in this file.
...
 [1.1.1]: https://github.com/sablier-labs/v2-periphery/compare/v1.1.0...v1.1.1
 [1.1.0]: https://github.com/sablier-labs/v2-periphery/compare/v1.0.3...v1.1.0
 [1.0.3]: https://github.com/sablier-labs/v2-periphery/compare/v1.0.2...v1.0.3
 [1.0.2]: https://github.com/sablier-labs/v2-periphery/compare/v1.0.1...v1.0.2
 [1.0.1]: https://github.com/sablier-labs/v2-periphery/compare/v1.0.0...v1.0.1
 [1.0.0]: https://github.com/sablier-labs/v2-periphery/releases/tag/v1.0.0
 
+## [1.2.0] - 2024-07-01
+
+### Changed
```
Comment (andreivladbrg) [inline CHANGELOG.md:17]: let's alphabetically order

File: CHANGELOG.md (hunk 2 above)
Comment (smol-ninja) [reply CHANGELOG.md:31]: Sure.

File: CHANGELOG.md (hunk 3 above)
Comment (smol-ninja) [reply CHANGELOG.md:28]: Its included in "Bump build dependencies ([#351](https://github.com/sablier-labs/v2-periphery/pull/351), [#283](https://github.com/sablier-labs/v2-periphery/pull/283))" section.

File: CHANGELOG.md (hunk 4 above)
Comment (smol-ninja) [reply CHANGELOG.md:22]: Yeah.

File: CHANGELOG.md (hunk 5 above)
Comment (smol-ninja) [reply CHANGELOG.md:17]: Good idea.

File: CHANGELOG.md (hunk 3 above)
Comment (andreivladbrg) [reply CHANGELOG.md:26]: but upgrading the solc version is not related to "dependencies"
`

Comment (smol-ninja) [reply CHANGELOG.md:26]: Understand. Then it should also be same in v2-core. I dont remember to mention solc version upgrade there.

Comment (smol-ninja) [reply CHANGELOG.md:26]: Feel free to push a commit to resolve these.

Comment (andreivladbrg) [reply CHANGELOG.md:26]: i've already pushed a commit in core

Comment (andreivladbrg) [reply CHANGELOG.md:26]: i will create a v1.2.0 github tag on core and the i will create another pr to bump the core which i will add to bump dependencies

**Review by andreivladbrg — APPROVED** [review]

File: package.json [hunk 6]
```diff
@@ -12,7 +12,7 @@
   "dependencies": {
     "@openzeppelin/contracts": "5.0.2",
     "@prb/math": "4.0.3",
-    "@sablier/v2-core": "github:sablier-labs/v2-core#staging"
+    "@sablier/v2-core": "github:sablier-labs/v2-core"
```
Comment (smol-ninja) [inline package.json:15]:
```suggestion
    "@sablier/v2-core": "1.2.0"
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -4,13 +4,39 @@ All notable changes to this project will be documented in this file.
...
 
+[1.2.0]: https://github.com/sablier-labs/v2-periphery/compare/v1.1.1...v1.2.0
 [1.1.1]: https://github.com/sablier-labs/v2-periphery/compare/v1.1.0...v1.1.1
 [1.1.0]: https://github.com/sablier-labs/v2-periphery/compare/v1.0.3...v1.1.0
 [1.0.3]: https://github.com/sablier-labs/v2-periphery/compare/v1.0.2...v1.0.3
 [1.0.2]: https://github.com/sablier-labs/v2-periphery/compare/v1.0.1...v1.0.2
 [1.0.1]: https://github.com/sablier-labs/v2-periphery/compare/v1.0.0...v1.0.1
 [1.0.0]: https://github.com/sablier-labs/v2-periphery/releases/tag/v1.0.0
 
+## [1.2.0] - 2024-07-02
```
Comment (smol-ninja) [inline CHANGELOG.md:15]: We should update this date too.

**Comment by smol-ninja** [comment]
Lezzz go 🚀🚀🚀

LGTM

**Pull request merged**